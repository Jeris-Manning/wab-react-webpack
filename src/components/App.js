import React from 'react';
import PropTypes from 'prop-types';

import ErrorBoundary from './ErrorBoundary';
import MyComponent from './MyComponent';

import 'semantic-ui-css/semantic.min.css';
import '../css/App.css';

const App = props => {
  return (
    <ErrorBoundary>
      <MyComponent title="WAB Widget" wab={props.wab} esriJS={props.esriJS} />
    </ErrorBoundary>
  );
};

App.propTypes = {
  wab: PropTypes.object,
  esriJS: PropTypes.object,
};

App.defaultProps = {
  wab: {},
  esriJS: {},
};

export default App;
