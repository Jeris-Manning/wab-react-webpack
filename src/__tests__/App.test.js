import React from 'react';
import { render, wait, cleanup } from 'react-testing-library';
import 'jest-dom/extend-expect';

import App from '../components/App';

const mock = {
  wab: {
    map: null,
    id: 0,
    config: {
      layerCollection: [],
    },
  },
  esriJS: {
    LayerInfos: jest.fn,
    FeatureLayer: jest.fn,
  },
};

afterEach(cleanup);

// https://testing-library.com/docs/react-testing-library/example-intro
describe('<App />', () => {
  it('renders the widget', async () => {
    // render component
    const { container, getByText, queryByText } = render(
      <App
        wab={mock.wab}
        esriJS={mock.esriJS}
        onOpen={jest.fn}
        onClose={jest.fn}
      />,
    );

    // test inital render
    expect(getByText('Loading...')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();

    // test after loading complete
    await wait(() =>
      expect(queryByText(/Loading\.\.\./i)).not.toBeInTheDocument(),
    );

    // this will be error bc we cant access the JS api from the test
    // but can see state move from loading
    expect(getByText('Error...')).toBeInTheDocument();
  });
});
