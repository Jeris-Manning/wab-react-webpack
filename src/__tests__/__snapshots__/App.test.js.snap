// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> displays entry to widget 1`] = `
<div
  className="sc-bdVaJa ldkUXY"
>
  <div
    className="ui header"
  >
    WAB Widget
  </div>
  <div
    className="ui segment"
  >
    <h3>
      WAB
    </h3>
  </div>
</div>
`;
